### DISCUSSION TITLE
Speed Estimator for Vehicle Tracking

### DISCUSSION THREAD

**Discussion by bthoma48** [roboflow/supervision#1248] [Q&A]
Where and how do I specifically add the configurations for both "vehicles.mp4" and "vehicles-result.mp4" in the ultralytics script, "ultralytics_example.py"?

Does it simply replace the code-line: "--source_video_path" and "--target_video-path"?

Can you specifically send the 146-line ultralytics script to incorporate "vehicles.mp4" and "vehicles-result.mp4"?

**Comment by LinasKo** [comment]
Hey @bthoma48 :wave: 

One minute, let me have a look.

**Reply by LinasKo** [reply]
I assume you mean the `examples/speed_estimation/ultralytics_example.py` file as that has 146 lines.

---

When you have the file on your machine, you're able to send arguments to it from your terminal as follows:
```bash
python ultralytics_example.py --source_video_path path/to/your/video.mp4 --target_video_path path/to/results/video.mp4
```

It will take the video, process it, and write the results into a new video, saving it in the location pointed to by the `--target_video_path` argument.

Note that the script assumes a certain location in the video is marked. You'll likely need to change variables `SOURCE`, `TARGET_WIDTH`, `TARGET_HEIGHT`.

For more modern tutorial, check out https://blog.roboflow.com/estimate-speed-computer-vision/

Hope this helps!

**Reply by bthoma48** [reply]
I appreciate the reply. Yes, that is the correct script. I’m relatively new to using (and navigating) Python, so apologize for the question. I was able to install ultralytics, pip, etc. within Python interpreter in addition to paste the 146-line ultralytics script into a new Python project file.

Would you be willing to send screenshots with the additional steps that you describe? I don’t quite understand where/how in Python I send those arguments.

**Reply by LinasKo** [reply]
I can help out, but the terminal screen might be differerent.

If you have a python file, how do you normally run it? Can you attach a screenshot? :slightly_smiling_face:

**Comment by bthoma48** [comment]
I don’t have access to the Python file right now, but here is an example screenshot from another Python project using the most recent version of PyCharm software. I simply click the green arrow (see in top right part of screen) to run the script:

![IMG_8152](https://github.com/roboflow/supervision/assets/171246123/3aabb8ff-5248-48ae-a723-458ca8e559be)

**Reply by LinasKo** [reply]
Ah, PyCharm. Take a peek here:
https://stackoverflow.com/questions/33102272/pycharm-and-sys-argv-arguments

It should either be `--source_video_path ./your_video.mp4 --target_video_path ./output.mp4` or `"--source_video_path" "./your_video.mp4" "--target_video_path" "./output.mp4"`

Make sure to replace `your_video.mp4` with the path to the video on your system.

**Reply by bthoma48** [reply]
Thank you. I’ll try it out later day and keep you posted if any issues. Appreciated.

**Reply by bthoma48** [reply]
Here is the ultralytics script (not edited) from: https://github.com/roboflow/supervision/blob/develop/examples/speed_estimation/ultralytics_example.py:
![image](https://github.com/roboflow/supervision/assets/171246123/1022d91e-4b10-45a4-ad7e-338fd3890049)

Here are the configurations requested from: https://colab.research.google.com/github/roboflow-ai/notebooks/blob/main/notebooks/how-to-estimate-vehicle-speed-with-computer-vision.ipynb?ref=blog.roboflow.com
![image](https://github.com/roboflow/supervision/assets/171246123/fc566092-a636-4668-8965-079d75365b50)

Here is the ultralytics script (edited with source video and target video paths for 'vehicles.mp4' for source video and 'vehicles-results.mp4' for target video) along with error messages"
![image](https://github.com/roboflow/supervision/assets/171246123/e23b660a-9fb6-4911-803f-a19d7807cfe7)

Do I enter the configurations for source video and target video paths in the ultralytics script code itself, or do I enter as run/debug "configuration"?
![image](https://github.com/roboflow/supervision/assets/171246123/5dffb056-970c-406b-a3d9-b9eca4336b74)
![image](https://github.com/roboflow/supervision/assets/171246123/baeb8228-1330-40c0-8ac9-993244176e93)
![image](https://github.com/roboflow/supervision/assets/171246123/29f19991-bb91-441e-bb76-5f72c00f45ef)

Again, I apologize for the questions, and I appreciate your help.

**Reply by LinasKo** [reply]
![pycharm_params](https://github.com/roboflow/supervision/assets/6500785/516969fb-1472-4de5-a968-dbf2d98e68fc)

Alternatively, it might be easier to edit the code.
On this line, you can set `source_video_path = "<path to your video">`.

Then, wherever `args.source_video_path` is mentioned, change it to `source_video_path` in the code.

If you do the same for every parameter you need, you won't need to pass any Script Parameters to PyCharm.

**Comment by bthoma48** [comment]
Unfortunately, I am receiving the following message when run Python in both PyCharm in addition to Google Colab. I used Gemini without success. I appreciate your help!

qt.qpa.xcb: could not connect to display :1qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "/usr/local/lib/python3.10/dist-packages/cv2/qt/plugins" even though it was found.This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem. Available platform plugins are: xcb.

**Reply by LinasKo** [reply]
This will be a tricky one, and a bit larger than what we have time for to help with.
QT is related to the UI your application might be trying to create.

I'd check if any of [this](https://stackoverflow.com/questions/71088095/opencv-could-not-load-the-qt-platform-plugin-xcb-in-even-though-it-was-fou) is helpful.

What platform are you trying to run on?
